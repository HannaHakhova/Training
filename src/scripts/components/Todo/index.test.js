import React from 'react';
import {shallow} from 'enzyme';
import Todo from './index';

describe('Todo component', () => {
    it('with empty todo object should match expected snapshot', () => {
        expect(shallow(<Todo toggle={() => {}} todo={{id: 1, text: 'Do task1', isDone: false}}/>)).toMatchSnapshot();
    });

    it('with not empty todo object should match expected snapshot', () => {
        expect(shallow(<Todo toggle={() => {}} todo={{id: 1, text: 'Do task1', isDone: true}}/>)).toMatchSnapshot();
    });
});

// jest.doMock('./Todo', () => {
//   const ComponentToMock = () => <div />;
//   return ComponentToMock;
// });

// const Todo = require('./Todo').default;
// // it('should display text of todo if todo.isDone is true', () => {
// //     const todo = {id: 1, text: 'Some text', isDone: true};
// //     const expectedResult = 'Some text';
// //     expect(expectedResult).toEqual(todo.text);
// //     });