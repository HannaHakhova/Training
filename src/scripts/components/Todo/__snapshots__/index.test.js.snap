// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component with empty todo object should match expected snapshot 1`] = `
<li>
  Do task1
  <span
    onClick={[Function]}
  >
    âœ“
  </span>
</li>
`;

exports[`Todo component with not empty todo object should match expected snapshot 1`] = `
<li>
  <del>
    Do task1
  </del>
  <span
    onClick={[Function]}
  >
    X
  </span>
</li>
`;
